diff --git a/cosmic-greeter-start.sh b/cosmic-greeter-start.sh
index 2e0cbf2..f72f2e5 100644
--- a/cosmic-greeter-start.sh
+++ b/cosmic-greeter-start.sh
@@ -1,3 +1,3 @@
 #!/bin/sh
 rm -rf /run/cosmic-greeter/cosmic/com.system76.CosmicSettingsDaemon/v1/*
-exec cosmic-comp systemd-cat -t cosmic-greeter cosmic-greeter
+exec cosmic-comp cosmic-greeter

diff --git a/cosmic-greeter.toml b/cosmic-greeter.toml
index 62072f4..596cba8 100644
--- a/cosmic-greeter.toml
+++ b/cosmic-greeter.toml
@@ -1,5 +1,5 @@
 [terminal]
-vt = "1"
+vt = "7"
 
 [general]
 service = "cosmic-greeter"

diff --git a/debian/cosmic-greeter-daemon.service b/debian/cosmic-greeter-daemon.service
index a567119..03c67db 100644
--- a/debian/cosmic-greeter-daemon.service
+++ b/debian/cosmic-greeter-daemon.service
@@ -1,11 +1,10 @@
-[Unit]
-Description=COSMIC Greeter Daemon
-
-[Service]
-Type=dbus
-BusName=com.system76.CosmicGreeter
-ExecStart=/usr/bin/cosmic-greeter-daemon
-Restart=on-failure
-
-[Install]
-WantedBy=multi-user.target
+#  SPDX-License-Identifier: GPL-3.0-or-later
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 3 as published by
+# the Free Software Foundation.
+ 
+[D-BUS Service]
+Name=com.system76.CosmicGreeter
+Exec=/usr/bin/cosmic-greeter-daemon &
+User=root
