diff --git a/libspa/src/param/video/raw.rs b/libspa/src/param/video/raw.rs
index ad18f9f..0c9334c 100644
--- a/libspa/src/param/video/raw.rs
+++ b/libspa/src/param/video/raw.rs
@@ -228,7 +228,6 @@ impl VideoInfoRaw {
     pub fn new() -> Self {
         Self(spa_sys::spa_video_info_raw {
             format: VideoFormat::Unknown.as_raw(),
-            flags: 0,
             modifier: 0,
             size: Rectangle {
                 width: 0,
@@ -257,20 +256,12 @@ impl VideoInfoRaw {
         VideoFormat::from_raw(self.0.format)
     }
 
-    pub fn set_flags(&mut self, flags: VideoFlags) {
-        self.0.flags = flags.bits();
-    }
-
-    pub fn flags(self) -> VideoFlags {
-        VideoFlags::from_bits_retain(self.0.flags)
-    }
-
     pub fn set_modifier(&mut self, modifier: u64) {
-        self.0.modifier = modifier;
+        self.0.modifier = modifier as i64;
     }
 
     pub fn modifier(self) -> u64 {
-        self.0.modifier
+        self.0.modifier as u64
     }
 
     pub fn set_size(&mut self, size: Rectangle) {
@@ -409,7 +400,6 @@ impl Debug for VideoInfoRaw {
 
         f.debug_struct("VideoInfoRaw")
             .field("format", &self.format())
-            .field("flags", &self.flags())
             .field("modifier", &self.modifier())
             .field("size", &self.size())
             .field("framerate", &self.framerate())
