diff --git a/data/start-cosmic b/data/start-cosmic
index f723d5e..d0ca899 100755
--- a/data/start-cosmic
+++ b/data/start-cosmic
@@ -55,6 +55,11 @@ if [ -z "$QT_QPA_PLATFORMTHEME" ]; then
     done
 fi
 
+SSH_AGENT=""
+if [ -x /usr/bin/ssh-agent ]; then
+  SSH_AGENT=/usr/bin/ssh-agent
+fi
+
 # Start gnome keyring components if the daemon is active
 # -> check if $XDG_RUNTIME_DIR/keyring exists
 if [ -d "$XDG_RUNTIME_DIR/keyring" ]; then
@@ -104,7 +109,7 @@ fi
 
 # Run cosmic-session
 if [ -z "${DBUS_SESSION_BUS_ADDRESS}" ]; then
-    exec /usr/bin/dbus-run-session -- /usr/bin/cosmic-session
+    exec /usr/bin/dbus-run-session -- $SSH_AGENT /usr/bin/cosmic-session
 else
-    exec /usr/bin/cosmic-session
+    exec $SSH_AGENT /usr/bin/cosmic-session
 fi
