diff --git a/data/start-cosmic b/data/start-cosmic
index f723d5e..d0ca899 100755
--- a/data/start-cosmic
+++ b/data/start-cosmic
@@ -43,6 +43,11 @@ if command -v systemctl >/dev/null; then
     systemctl --user import-environment XDG_SESSION_TYPE XDG_CURRENT_DESKTOP DCONF_PROFILE
 fi
 
+SSH_AGENT=""
+if [ -x /usr/bin/ssh-agent ]; then
+  SSH_AGENT=/usr/bin/ssh-agent
+fi
+
 # Start gnome keyring components if the daemon is active
 # -> check if /run/user/$UID/keyring exists
 if [ -d "/run/user/$(id -u)/keyring" ]; then
@@ -60,7 +65,7 @@ fi
 
 # Run cosmic-session
 if [[ -z "${DBUS_SESSION_BUS_ADDRESS}" ]]; then
-    exec /usr/bin/dbus-run-session -- /usr/bin/cosmic-session
+    exec /usr/bin/dbus-run-session -- $SSH_AGENT /usr/bin/cosmic-session
 else
-    exec /usr/bin/cosmic-session
+    exec $SSH_AGENT /usr/bin/cosmic-session
 fi
